query/src/storages/random/random_table.rs

```diff
@@ -0,0 +1,188 @@
+use crate::pipelines::new::NewPipe;
+use crate::pipelines::new::NewPipeline;
+use crate::sessions::QueryContext;
+use crate::storages::StorageContext;
+use crate::storages::StorageDescription;
+use crate::storages::Table;
+
+pub struct RandomTable {
+    table_info: TableInfo,
+}
+
+impl RandomTable {
+    pub fn try_create(_ctx: StorageContext, table_info: TableInfo) -> Result<Box<dyn Table>> {
+        Ok(Box::new(Self { table_info }))
+    }
+
+    pub fn description() -> StorageDescription {
+        StorageDescription {
+            engine_name: "RANDOM".to_string(),
+            comment: "RANDOM Storage Engine".to_string(),
+            ..Default::default()
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Table for RandomTable {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn get_table_info(&self) -> &TableInfo {
+        &self.table_info
+    }
+
+    async fn read_partitions(
+        &self,
+        _ctx: Arc<QueryContext>,
+        _push_downs: Option<Extras>,
+    ) -> Result<(Statistics, Partitions)> {
+        Ok((Statistics::default(), vec![]))
+    }
+
+    fn benefit_column_prune(&self) -> bool {
+        true
+    }
+
+    fn read2(
+        &self,
+        ctx: Arc<QueryContext>,
+        plan: &ReadDataSourcePlan,
+        pipeline: &mut NewPipeline,
+    ) -> Result<()> {
+        let output = OutputPort::create();
+        let schema = self.table_info.schema();
+        pipeline.add_pipe(NewPipe::SimplePipe {
+            inputs_port: vec![],
+            outputs_port: vec![output.clone()],
+            processors: vec![RandomSource::create(
+                ctx,
+                output,
+                schema,
+                plan.push_downs.clone(),
+            )?],
+        });
+
+        Ok(())
+    }
+}
+
+struct RandomSource {
+    schema: DataSchemaRef,
+    /// The number of rows needed to generate.
+    limit: usize,
+    /// record rows count.
+    rows: usize,
+    /// The max number of rows can one `generate` generate.
+    block_size: usize,
+}
+
+impl RandomSource {
+    pub fn create(
+        ctx: Arc<QueryContext>,
+        output: Arc<OutputPort>,
+        schema: DataSchemaRef,
+        extras: Option<Extras>,
+    ) -> Result<ProcessorPtr> {
+        let block_size = ctx.get_settings().get_max_block_size()? as usize;
+        let mut output_schema = schema;
+        // If extras.push_downs is None or extras.push_down.limit is None,
+        // set limit to `max_block_size`.
+        let limit = match extras {
+            Some(push_downs) => {
+                if let Some(projection) = push_downs.projection {
+                    output_schema = Arc::new(output_schema.project(projection));
+                }
+                match push_downs.limit {
+                    Some(limit) => limit,
+                    None => block_size,
+                }
+            }
+            None => block_size,
+        };
+
+        SyncSourcer::create(ctx, output, RandomSource {
```

Comment:
Should I implement a multi sourcer? Like:

https://github.com/datafuselabs/databend/blob/652bb413c7e5277928d7c9110ca7198bb9098023/query/src/storages/memory/memory_table.rs#L225-L238

Reply:
Yes,better to have.

Reply:
If so, how many rows should one source generate one time? Will the sources generate rows parallelly or concurrently?

Reply:
You can split the total rows into n Parts where n is the num of threads.

refer:  read_partitions  function  in https://github.com/datafuselabs/databend/blob/9261d9cc06d959fdca5516de7cd1381f5c9509df/query/src/table_functions/numbers_table.rs